site/_posts/2017-09-29-miqldap-to-sssd.md

```diff
@@ -0,0 +1,315 @@
+---------------------------------------------------------------------
+
+The *miqldap_to_sssd* will reconfigure the appliance for authentication **Mode: External (httpd)**
+
+[![Appliance Configuration Mode: External (httpd)](/assets/images/blog/miqldap_to_sssd_auth_config_mode_ext_auth.png)](/assets/images/blog/miqldap_to_sssd_auth_config_mode_ext_auth.png)
+
+**The other miqldap_to_sssd arguments**
+=====================================================================
+
+So far only 3 of the 5 arguments supported by *miqldap_to_sssd* have been used.
+The other 2 will be explained here.
+
+Again the 5 supported argument are:
+
+<pre><code style="white-space: pre">
+miqldap_to_sssd --help
+Usage: ruby tools/miqldap_to_sssd.rb [options]
+  -d, --basedn-domain=                       The Base DN domain name, e.g. example.com
+  -c, --tls-cacert=                          Path to certificate file
+  -n, --only-change-userids                  normalize the userids then exit
+  -s, --skip-post-coversion-userid-change    Do the MiqLdap to SSSD conversion but skip the normalizing of the userids
+  -h, --help                                 Show this message
+</code></pre>
+
+As mentioned earlier :
+
+When the **LDAP Host** is an LDAP directory the username field stored in the MiQ database is a **Distinguished Name**
+
+When the **LDAP Host** is an AD directory the username field stored in the MiQ database is a **User Principal Name**.
+
+After the conversion to external authentication the username field stored in the MiQ database will always be **User Principal Name**.
+
+To ensure that the username field stored in the MiQ database will always be **User Principal Name** the *miqldap_to_sssd* conversion
+tool will convert the username fields to **User Principal Name** if they are not.
+
+If for any reason this conversion of the username field to  **User Principal Name** is not desired the --skip-post-coversion-userid-change
+argument can be used.
+
+If, on the other hand, the conversion from authentication **Mode: LDAP(s):** to **Mode: External (httpd)**
+is not desired the username userids can still be normalized to **User Principal Name** format by using the **--only-change-userids** argument.
+
+**Debugging Failed Conversions**
+=====================================================================
+
+There are many different possible authentication configurations. Some may be encountered that result in the *miqldap_to_sssd* conversion tool
+not fully completing. If things go wrong, to diagnose why, start by examining the log file */var/www/miq/vmdb/log/miqldap_to_sssd.log* 
+
+For more help diagnosing failures see the instructions for manually configuring an MiQ Appliance for external authentication.
+[(as documented here)](http://manageiq.org/docs/reference/latest/auth/ldap)
+
+This document has diagnostics tips, including:
+  - How to use **ldapsearch**, with usage examples.
+  - Hot to use **dbus-send**, with usage examples
```

Comment:
Didn't know dbus-send is "hot to use" ;)

Reply:
LOL! (I actually did LOL!) Good catch.